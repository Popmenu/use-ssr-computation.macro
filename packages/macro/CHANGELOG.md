# Change Log

## [1.2.4] - 2023-10-02
* Support Subscription computations
* Drop support for async computations

## [1.2.0] - 2023-08-28
* Support async computations
* Add support for error handling

## [1.1.0] - 2023-08-06
* Add support for `skip` option that skips the computation

## [1.0.3] - 2023-07-21
* Split the package into `macro` and `runtime` packages

[1.2.4]: https://github.com/shakacode/use-ssr-computation.macro/compare/%40popmenu/use-ssr-computation.macro%401.2.3...%40popmenu/use-ssr-computation.macro%401.2.4
[1.2.0]: https://github.com/shakacode/use-ssr-computation.macro/pull/51/files
[1.1.0]: https://github.com/shakacode/use-ssr-computation.macro/pull/40
[1.0.3]: https://github.com/shakacode/use-ssr-computation.macro/tree/f4edc2717a19a0a332db5cc679c0d536253d19ab
