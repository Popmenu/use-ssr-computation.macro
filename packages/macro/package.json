{
  "name": "@popmenu/use-ssr-computation.macro",
  "version": "1.1.0",
  "main": "lib/index.macro.js",
  "types": "lib/index.macro.d.ts",
  "license": "MIT",
  "scripts": {
    "build": "tsc -p .",
    "build:watch": "yarpm build --watch",
    "test": "jest",
    "compile-example": "yarn clear:babel-cache && babel --plugins=babel-plugin-macros --presets=@babel/preset-typescript example/index.ts",
    "clear:babel-cache": "rimraf -rf ./node_modules/.cache/babel-loader/*",
    "clean": "rimraf lib",
    "prepublishOnly": "yarpm clean && yarpm build"
  },
  "files": [
    "lib"
  ],
  "homepage": "https://github.com/Popmenu/use-ssr-computation.macro",
  "bugs": {
    "url": "https://github.com/Popmenu/use-ssr-computation.macro/issues"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/Popmenu/use-ssr-computation.macro.git"
  },
  "author": "Roman Kuksin <rkuksin.cpp@gmail.com>",
  "dependencies": {
    "@popmenu/use-ssr-computation.runtime": "*",
    "babel-plugin-macros": "^3.0.0"
  },
  "devDependencies": {
    "@babel/cli": "^7.8.4",
    "@babel/preset-typescript": "^7.22.5",
    "@types/babel-plugin-macros": "^3.1.0",
    "@types/babel__core": "^7.1.3",
    "@types/jest": "^29.5.2",
    "@types/node": "^20.3.1",
    "babel-plugin-tester": "^11.0.4",
    "jest": "^29.3.1",
    "rimraf": "^5.0.1",
    "ts-jest": "^29.1.0",
    "tslint": "^6.1.3",
    "typescript": "^5.1.3",
    "yarpm": "^1.2.0"
  },
  "keywords": [
    "SSR",
    "performance",
    "babel",
    "macro",
    "javascript",
    "typescript"
  ],
  "jest": {
    "transform": {
      "^.+\\.tsx?$": "ts-jest"
    }
  }
}
