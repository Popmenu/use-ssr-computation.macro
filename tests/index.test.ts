const pluginTester = require("babel-plugin-tester").default;
const plugin = require("babel-plugin-macros");

pluginTester({
  pluginOptions: {
    useSSRComputation: {
      side: "client",
    },
  },
  plugin,
  snapshot: true,
  babelOptions: {
    filename: __filename,
  },
  tests: {
    "no usage": `import { useSSRComputation } from "../lib/index.macro"`,
    "server-side": `
      import { useSSRComputation } from "../lib/index.macro"
      const x = useSSRComputation("./a.ssr-computation");
    `,
    "multiple-imports": `
      import { useSSRComputation } from "../lib/index.macro"
      const x = useSSRComputation("./a.ssr-computation")
      const y = useSSRComputation("./a.ssr-computation")
    `,
  },
});

pluginTester({
  beforeAll: () => {
    process.env['HEEH'] = 'CLIENT_SIDE';
  },
  pluginOptions: {
    useSSRComputation: {
      side: "server",
    },
  },

  plugin,
  restartTitleNumbering: true,
  snapshot: true,
  babelOptions: {
    filename: __filename,
  },
  tests: {
    "client-side": `
      import { useSSRComputation } from "../lib/index.macro"
      const x = useSSRComputation("./a.ssr-computation")
    `,
  },
});
