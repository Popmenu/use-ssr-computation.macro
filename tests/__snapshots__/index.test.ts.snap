// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros client-side: client-side 1`] = `

import { useSSRComputation } from "../lib/index.macro"
const x = useSSRComputation("./a.ssr-computation")

      ↓ ↓ ↓ ↓ ↓ ↓

import __a from "./a.ssr-computation";
import { useSSRComputationServer } from "use-ssr-computation/lib/useSSRComputation";
const x = useSSRComputationServer("./a.ssr-computation");


`;

exports[`macros multiple-imports: multiple-imports 1`] = `

import { useSSRComputation } from "../lib/index.macro"
const x = useSSRComputation("./a.ssr-computation")
const y = useSSRComputation("./a.ssr-computation")

      ↓ ↓ ↓ ↓ ↓ ↓

import { useSSRComputationClient } from "use-ssr-computation/lib/useSSRComputation";
const x = useSSRComputationClient("./a.ssr-computation");
const y = useSSRComputationClient("./a.ssr-computation");


`;

exports[`macros no usage: no usage 1`] = `

import { useSSRComputation } from "../lib/index.macro"

      ↓ ↓ ↓ ↓ ↓ ↓



`;

exports[`macros server-side: server-side 1`] = `

import { useSSRComputation } from "../lib/index.macro"
const x = useSSRComputation("./a.ssr-computation");

      ↓ ↓ ↓ ↓ ↓ ↓

import { useSSRComputationClient } from "use-ssr-computation/lib/useSSRComputation";
const x = useSSRComputationClient("./a.ssr-computation");


`;
